lib/mastodon/upgrade_cli.rb

```diff
@@ -41,23 +41,32 @@ def storage_schema
         klass.find_each do |record|
           attachment_names.each do |attachment_name|
             attachment = record.public_send(attachment_name)
+            upgraded   = false
 
             next if attachment.blank? || attachment.storage_schema_version >= CURRENT_STORAGE_SCHEMA_VERSION
 
-            attachment.styles.each_key do |style|
-              case Paperclip::Attachment.default_options[:storage]
-              when :s3
-                upgrade_storage_s3(progress, attachment, style)
-              when :fog
-                upgrade_storage_fog(progress, attachment, style)
-              when :filesystem
-                upgrade_storage_filesystem(progress, attachment, style)
+            styles = attachment.styles.keys
+
+            styles << :original unless styles.include?(:original)
+
+            styles.each do |style|
+              success = begin
+                case Paperclip::Attachment.default_options[:storage]
+                when :s3
+                  upgrade_storage_s3(progress, attachment, style)
+                when :fog
+                  upgrade_storage_fog(progress, attachment, style)
+                when :filesystem
+                  upgrade_storage_filesystem(progress, attachment, style)
+                end
```

Comment:
Now it will report success if the last style gets moved, but not if there is any error, and especially not if an error occurs on the `:original` style but not on another one.

Reply:
I think you are possibly mistaken? I've renamed some variables. The `success` tracks the success of the current, given style only. Then the `upgraded` variable is only set when the move succeeded and it was the `:original` style

Reply:
Oh, right. I just thought there was a global return value for this, and looked at another part of the code.